python_modules/dagster/dagster_tests/definitions_tests/test_asset_check_selection.py

```diff
@@ -0,0 +1,162 @@
+from dagster import (
+    AssetCheckResult,
+    AssetSelection,
+    Definitions,
+    ExecuteInProcessResult,
+    asset,
+    asset_check,
+    define_asset_job,
+)
+from dagster._core.definitions.asset_check_spec import AssetCheckHandle
+from dagster._core.definitions.unresolved_asset_job_definition import UnresolvedAssetJobDefinition
+
+
+@asset
+def asset1():
+    ...
+
+
+@asset
+def asset2():
+    ...
+
+
+@asset_check(asset=asset1)
+def asset1_check1():
+    return AssetCheckResult(success=True)
+
+
+@asset_check(asset=asset1)
+def asset1_check2():
+    return AssetCheckResult(success=True)
+
+
+@asset_check(asset=asset2)
+def asset2_check1():
+    return AssetCheckResult(success=True)
+
+
+def execute_asset_job_in_process(asset_job: UnresolvedAssetJobDefinition) -> ExecuteInProcessResult:
+    assets = [asset1, asset2]
+    asset_checks = [asset1_check1, asset1_check2, asset2_check1]
+    defs = Definitions(assets=assets, jobs=[asset_job], asset_checks=asset_checks)
+    job_def = defs.get_job_def(asset_job.name)
+    return job_def.execute_in_process()
+
+
+def test_job_with_all_checks_no_materializations():
+    job_def = define_asset_job("job1", selection=AssetSelection.all_asset_checks())
+    result = execute_asset_job_in_process(job_def)
+    assert result.success
+
+    assert len(result.get_asset_materialization_events()) == 0
+    check_evals = result.get_asset_check_evaluations()
+    assert {check_eval.asset_check_handle for check_eval in check_evals} == {
+        AssetCheckHandle(asset1.key, "asset1_check1"),
+        AssetCheckHandle(asset1.key, "asset1_check2"),
+        AssetCheckHandle(asset2.key, "asset2_check1"),
+    }
+
+
+def test_job_with_all_checks_for_asset():
+    job_def = define_asset_job("job1", selection=AssetSelection.asset_checks_for_assets(asset1))
+    result = execute_asset_job_in_process(job_def)
+    assert result.success
+
+    assert len(result.get_asset_materialization_events()) == 0
+    check_evals = result.get_asset_check_evaluations()
+    assert {check_eval.handle for check_eval in check_evals} == {
+        AssetCheckHandle(asset1.key, "asset1_check1"),
+        AssetCheckHandle(asset1.key, "asset1_check2"),
+    }
+
+
+def test_job_with_asset_and_all_its_checks():
+    job_def = define_asset_job("job1", selection=AssetSelection.assets(asset1) & AssetSelection.asset_checks_for_assets(asset1))
```

Comment:
Using set operators is super confusing since this doesn't act like a set '&'. This really is an OR/UNION isn't it. Why don't we just do that instead of being cute.

Reply:
Meaning that if this really following set semantics this wolud be a null job